agents-for-bedrock/features-examples/08-create-agent-with-guardrails/08-create-agent-with-guardrails.ipynb

```diff
@@ -0,0 +1,1956 @@
+    "print(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e59b3614",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "##### Invoke Agent to execute function from Action Group\n",
+    "Now let's test our Action Group functionality and create a new reservation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e0201c0-abb3-4ad5-a7e7-1750d14a2717",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "query = \"Hi, I am Anna. I want to create a banking appointment for 8pm on the 5th of May 2024.\"\n",
+    "response = invokeAgent(query, session_id)\n",
+    "print(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7043ba4",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "##### Invoke Agent with prompt attribute\n",
+    "\n",
+    "Great! We've used our agent to do the first reservation. However, often when booking tables in restaurants we are already logged in to systesm that know our names. How great would it be if our agent would know it as well?\n",
+    "\n",
+    "To do so, we can use the session context to provide some attributes to our prompt. In this case we will provide it directly to the prompt using the [`promptSessionAttributes`](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-session-state.html) parameter. Let's also start a new session id so that our agent does not memorize our name."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "139c210f",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "session_id:str = str(uuid.uuid1())\n",
+    "query = \"Should I invest in bitcoin?\"\n",
+    "session_state = {\n",
+    "    \"promptSessionAttributes\": {\n",
+    "        \"name\": \"John\"\n",
+    "    }\n",
+    "}\n",
+    "response = invokeAgent(query, session_id, session_state=session_state)\n",
+    "print(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2cf0d98c",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## 6. Creating and associating a Amazon Bedrock Guardrail\n",
+    "The response above indirectly provided the investment advice about bitcoin in our example document to the user, however as a banking organization we cannot be seen to give any investment advice to the user. As a result, lets create a Bedrock Guardrail to prevent the user from doing this"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bfb8612f-6221-4199-b44b-4c341ac08f7f",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "response = bedrock_client.create_guardrail(\n",
```

Comment:
needs error handling

`ConflictException: An error occurred (ConflictException) when calling the CreateGuardrail operation: Another guardrail in your account already has this name. Retry your request with a different name.`